File: 0/0/4/004.py
# Copyright (C) 2013 Jorge Aparicio


def is_palindrome(n):
    s = str(n)
    return s == s[::-1]

print(max([a * b
      for a in range(100, 1000)
      for b in range(a, 1000)
      if is_palindrome(a * b)]))

File: 0/0/4/004.rs
// Copyright (C) 2013 Jorge Aparicio

fn main() {
  let mut max = 0;

  for a in range(100, 1000) {
    for b in range(100, a) {
      let p = a * b;

      if p > max && isPalindrome(p) {
        max = p;
      }
    }
  }

  println!("{}", max);
}

fn isPalindrome(n: int) -> bool {
  let (mut reversed, mut tmp) = (0, n);

  while tmp != 0 {
    reversed = 10*reversed + tmp%10;
    tmp /= 10;
  }

  reversed == n
}

